MdeModulePkg/Universal/PlatformDriOverrideDxe/PlatDriOverrideLib.c

```diff
@@ -361,7 +361,7 @@ UpdateFvFileDevicePath (
     // Build the shell device path
     //
     NewDevicePath = DevicePathFromHandle (FoundFvHandle);
-    if (NewDevicePath != NULL) {
+    if (NewDevicePath == NULL) {

```

Comment:
Why not ASSERT here? If intentional, presumably the idea must be that this case is likely/possible with reasonable input, but if so, can you briefly explain why so?